### DISCUSSION TITLE
nvidia support

### DISCUSSION THREAD

**Discussion by mdesson** [Manjaro-Sway/manjaro-sway#188] [Q&A] [answered]
I am trying to install manjaro sway edition following the instructions (downloaded both files and `cat`'d them together). I selected open source drivers on the live iso. When boots off of the usb key the CPU remains at 98% or 99% consistently.

I'm unable to install as it is so slow it has difficulty picking up key presses accurately (it will interpret a single key down as several, making it nearly impossible to log into wifi).

**Version**: Latest release (202110130235)
**CPU**: i7-6700
**GPU**: RTX 3070
**RAM**: 16GB

![20211013_084028](https://user-images.githubusercontent.com/25888319/137136997-ea274a4d-7234-437a-b74e-8ebcc8d36588.jpg)

**Comment by boredland** [comment]
So I think/guess that nvidia open source drivers still don't play well with sway (see our readme) and it is falling back to strange CPU based renders or sth. like that. The proprietary drivers seem to work better recently, but I did not confirm that myself and due to a lack of Nvidia GPU wont be able to confirm that. What I can confirm: the current image works as expected in a Gnome Boxes VM.

**Comment by mdesson** [comment]
Thanks for the quick response. I think you're right that's trying some funky CPU rendering that it's trying to pass through the GPU (since that's where the monitor is plugged in). 

A heads up that [this](https://github.com/Manjaro-Sway/manjaro-sway/issues/140) issue referenced in the readme still occurs on the proprietary drivers (at least on my GPU). Same logs from `zsh`, etc. Setting the env var `WLR_DRM_NO_MODIFIERS` to `1` in zsh (as recommended in the issue and in the archwiki link) does not fix it either, so I'm unable to install.

I'm not knowledgeable enough to troubleshoot and put up a PR to fix this myself, but if you'd like I can try some things thing and share the logs to help you debug and fix. For now, I can say that it's identical to the github issue (same output on `sway -d`) with the exception that `export WLR_DRM_NO_MODIFIERS=1` and then running `sway` in zsh did not work.

**Comment by boredland** [comment]
While I don't think there is a lot we can do here "downstream" i think we could and should keep this thread open, since you won't be the last running into it.

**Comment by mdesson** [comment]
Works for me! I'll check back in if at some point I find out that it support for nvidia GPUs has improved. Thanks again.

**Comment by 825i** [comment]
Gonna give this a shot myself on the Nvidia Quadro T500.

EDIT: Haven't been able to seeing as Wayland doesn't seem to support it properly at all, so I would need to swap to XWayland which got support for this a few days ago: https://lists.x.org/archives/xorg/2021-October/060798.html

**Comment by boredland** [comment]
@pythonInRelay you're misunderstanding the purpose of xwayland. it runs x containers as wayland children. it is not an 'alterantive to' wayland, but an application to be run on wayland. and we already ship with xwayland anyways.

**Comment by boredland** [comment, marked as answer]
according to [this](https://forums.developer.nvidia.com/t/nvidia-495-on-sway-tutorial-questions-arch-based-distros/192212) thread on the nvidia forums, a lot of this is still extremely experimental.

**Comment by boredland** [comment]
...and apparently one needs to run sway on proprietary drivers with `--unsupported-gpu`

**Comment by 825i** [comment]
Hi @boredland amazing job providing that helpful link. Indeed very experimental but still otherwise a positive and hopeful situation that shows good support for Sway is really not as far off as I think we all previously thought.

Really seems like the main bottleneck here is still wlroots/sway dev and his personal hatred for Nvidia :/ really hope that we can get around that soon, otherwise the next step might be forking Sway and taking it in a more "Nvidia friendly" direction.

> ...and apparently one needs to run sway on proprietary drivers with `--unsupported-gpu`

I think the flag was `--my-next-gpu-wont-be-nvidia` :D

**Comment by boredland** [comment]
Sooo, did sb try `unsupported-gpu` with current 1.7 sway and wl-roots?
